geopandas/testing.py

```diff
@@ -92,16 +92,15 @@ def assert_geoseries_equal(
     """
     assert len(left) == len(right), "%d != %d" % (len(left), len(right))
 
-    msg = "dtype should be a GeometryDtype, got {0}"
-    assert isinstance(left.dtype, GeometryDtype), msg.format(left.dtype)
-    assert isinstance(right.dtype, GeometryDtype), msg.format(left.dtype)
+    if check_dtype:
```

Comment:
If we make this change, I think it should rather be the default (`check_dtype=True`) ? 

Although, you can't have GeoSeries without having a geometry dtype normally. So since we still check that by default, it maybe doesn't matter much in practice

Reply:
> If we make this change, I think it should rather be the default (check_dtype=True) ?

True, but wouldn't it be a breaking change? This is a part of public API. 

I did this change so you can do `assert_geoseries_equal(Series, GeoSeries, check_series_type=False)`, which might not be needed.

Reply:
> I did this change so you can do assert_geoseries_equal(Series, GeoSeries, check_series_type=False)

Ah, yes, as otherwise you would have to do `check_series_type=False, check_dtype=False` when comparing to an object Series. That makes sense.

Reply:
Although, you can have a Series with geometry type, so there might still be value in only not checking the Series subclass type ..

> True, but wouldn't it be a breaking change? This is a part of public API.

Yes, that true. But, it would align it with pandas' assert_series_equal (not directly sure what's the historical reason for having a different default here in geopandas), and I think it would only matter since the introduction of the geometry dtype (so 0.6, still relatively recent) ? As before it was simply always object dtype ?

Reply:
I think we should be able to check GeoSeries with both following cases.

```py
# series has GeometryDtype
a = geopandas.array.from_shapely([
         Polygon([(0, 2), (0, 0), (2, 0), (2, 2)]),
         Polygon([(2, 2), (4, 2), (4, 4), (2, 4)]),
     ])
s1 = pd.Series(a)
```

```py
# series has object dtype
s = Series(
     [
         Polygon([(0, 2), (0, 0), (2, 0), (2, 2)]),
         Polygon([(2, 2), (4, 2), (4, 4), (2, 4)]),
     ]
 )
```
The second one currently fails on `assert isinstance(left.dtype, GeometryDtype), msg.format(left.dtype)` and there's no way around, `check_series_type=False, check_dtype=False` does not help. So it make sense to wrap it under `if check_dtype` being `True` by default.

This line `assert left.dtype == right.dtype, "dtype: %s != %s" % (left.dtype, right.dtype)` seems to be redundant, or is there a case when `isinstance(left.dtype, GeometryDtype)` is True for both series where `left.dtype != right.dtype`?

Reply:
> So it make sense to wrap it under if check_dtype

Yes, with that I fully agree (as otherwise the `check_dtype` argument has no effect)

>  is there a case when isinstance(left.dtype, GeometryDtype) is True for both series where left.dtype != right.dtype?

I don't think so. Right now, our geometry dtype is not parametrized, so all instances are always equal to each other (in principle, we could add a geometry type, or the crs, .. to the dtype, and then not all geometry dtypes would be equal)

Reply:
Okay. I changed it accordingly and added a test to cover both situations outlined above.